### PR TITLE
[Bug fix] Redid the package argument parsing

### LINKED ISSUE #15: Stuart_ci_build -p parameter eats all positional parameters following it

### PR DESCRIPTION
**Pull request by matthewfcarlson**
Fixes #15 but basically the argument parsing for -p was using nargs and consuming other arguments. A few solutions were proposed and we elected to use csv and multiple -p arguments

### CODE REVIEW

**Comment by matthewfcarlson** [comment]
Alright this should be good to go.

File: edk2toolext/invocables/edk2_ci_build.py [hunk 1]
```diff
@@ -113,14 +113,21 @@ def GetLoggingLevel(self, loggerType):
...
     def AddCommandLineOptions(self, parser):
-
-        parser.add_argument('-p', '--pkg', '--pkg-dir', dest='packageList', nargs="+", type=str,
-                            help='A package or folder you want to test (abs path or cwd relative).  '
-                            'Can list multiple by doing -p <pkg1> <pkg2> <pkg3>', default=[])
+        '''  Add command line options to the argparser '''
+        # This will parse the packages that we are going to build
+        parser.add_argument('-p', '--pkg', '--pkg-dir', dest='packageList', type=str,
+                            help='A package or folder you want to test (abs path or cwd relative).'
+                            'Can list multiple by doing -p <pkg1>,<pkg2> or -p <pkg3> -p <pkg4>. It must end in Pkg.',
```
Comment (spbrogan) [inline edk2toolext/invocables/edk2_ci_build.py]: I don't think we can require it ending in Pkg

File: edk2toolext/invocables/edk2_ci_build.py [hunk 2]
```diff
@@ -113,14 +113,21 @@ def GetLoggingLevel(self, loggerType):
...
+                            action="append")
 
     def RetrieveCommandLineOptions(self, args):
         '''  Retrieve command line options from the argparser '''
-        self.packageList = args.packageList
+        packageListSet = set()
+        for item in args.packageList:  # Parse out the individual packages
+            item_list = item.split(",")
```
Comment (spbrogan) [inline edk2toolext/invocables/edk2_ci_build.py]: I would suggest using strip() as well to make sure there is no whitespace at beginning or end.

**Comment by matthewfcarlson** [comment]
Fixed

**Pull request merged**